nntrainer/graph/network_graph.cpp

```diff
@@ -681,16 +681,25 @@ void NetworkGraph::inPlaceOptimize(Manager &manager) {
   }
 }
 
-void NetworkGraph::init2runContext(InitLayerContext &init_context,
-                                   RunLayerContext &run_context) {
-  // NOTE: this just create the wrappers and does not actually memory inside
-  // these wrappers
-  // TODO: create wrappers for weights - initialize is already done, so the
-  // object creation can be done
-  // TODO: create wrapper for the temporary tensors
-  // TODO: create wrappers for outputs
-  // TODO: create a new context with these new wrappers and then copy assign
-  // run_context
+void NetworkGraph::updateRunContext(std::shared_ptr<Manager> &manager,
+                                    const std::shared_ptr<LayerNode> &lnode) {
+  /**
+   * using copy assignment allows setting run_context without adding more
+   * interfaces
+   */
+  const GraphNode &gnode = *lnode.get();
+  const InitLayerContext &init_context = lnode->getInitContext();
+  /**
+   * @todo must use existing properties like name/trainable of run_context to
+   * create the new run_context
+   */
+  // const RunLayerContext &run_context = lnode->getRunContext();
+
+  lnode->updateRunContext(RunLayerContext(
```

Comment:
Update sounds like it's updating part of the context. how about `swapRunContext` ?

Reply:
or how about sth like moving `creatingRunContext to initContext`?
```cpp
RunContext InitContext::CreateRunContext(Manager manager);
void LayerNode::initRunContext(Manager manager) {
   this->runContext = initContext.CreateRunContext(manager) 
}
```

Reply:
> Update sounds like it's updating part of the context. how about `swapRunContext` ?

Actually it is returning updated RunContext. RunContext does not has any interface to update it (to protect from users). So, only way to update is new RunContext.

Reply:
aah, i understand the confusion. This interface is temporary, and updated in #1301. Please check it once.

Reply:
Thanks :)